### ISSUE TITLE
Widget.js jQuery include breaks cart page

### ISSUE THREAD

**Issue by MichaelThessel** [amzn/amazon-payments-magento-plugin#306]
Hi,

I'm currently running 1.4.2 and I get errors about the jQuery.on() function missing on the cart page. I traced this down to the amazon payments plugin including jQuery 1.4 through Widgets.js and overwriting the already existing jQuery. 

I saw that a similar problem was discussed in #284 but in our current install Widgets.js is only included once.

**Comment by MichaelThessel** [comment]
You can reproduce this on the cart page here: https://www.andrewchristian.com/

**Comment by snoroozi** [comment]
I never received a clear answer on this.

**Comment by danielneu** [comment]
Hi,

I had a quick look at the cart page, please see the screenshot attached.

![firebug - 2017-02-16 07 54 30](https://cloud.githubusercontent.com/assets/10086275/23010741/6cd71264-f41d-11e6-96d0-c81cb5f8ce3b.png)

The file is loaded twice, @islandskater43 added some hints in the ticket you are referencing to get rid of it.
Could you try if this helps?

Best wishes
Daniel

**Comment by MichaelThessel** [comment]
Correct, not sure why I missed that the first time.

I dug into this deeper and found that the payment script include (amazon_payments/script.phtml) for the mini cart and the dedicated payment script include on the cart page conflict.

This diff solved the problem for me:

```
diff --git a/app/design/frontend/base/default/layout/amazon_payments.xml b/app/design/frontend/base/default/layout/amazon_payments.xml
index 9c0ddb5b2..a049db5fd 100644
--- a/app/design/frontend/base/default/layout/amazon_payments.xml
+++ b/app/design/frontend/base/default/layout/amazon_payments.xml
@@ -23,17 +23,6 @@
             <action method="addItem"><type>skin_css</type><name>amazon_payments/css/popup_alphacube.css</name><params/></action>
         </reference>
 
-        <reference name="checkout.cart">
-            <block type="amazon_payments/script" name="amazon_payments.script" template="amazon_payments/script.phtml" />
-
-            <!-- ifconfig only works with action -->
-            <reference name="before_body_end">
-                <action method="append" ifconfig="payment/amazon_payments/enabled">
-                    <block>amazon_payments.script</block>
-                </action>
-            </reference>
-
-        </reference>
 
        <!-- To Move Amazon Payments button around on the cart page, change the after="" attribute to before="" in the below block tags -->
```

Obviously this is not ideal. I'm not sure why you would load the payment script include twice in the first place. As several other people are reporting this issue, it might be a good idea to address this issue in general.

While looking into this several other things stood out.

Widgets.js accounts for 273kb. I don't see why you need to include a JS lib like jQuery into what is supposed to be a lightweight integration.

Also, I constantly get warnings about how Widget.js is loaded. You might want to address this too.

> A Parser-blocking, cross-origin script, https://api-cdn.amazon.com/sdk/login1.js, is invoked via document.write. This may be blocked by the browser if the device has poor network connectivity. See https://www.chromestatus.com/feature/5718547946799104 for more details.

**Comment by danielneu** [comment]
Thanks for the valuable feedback,

the Widgets.js size is nothing which is related to this module, but I forwarded it to the relevant people,

All other concerns are valid as well and will be addressed,

The warning, which Chrome is throwing, can be removed by asynchronously loading our js files, which is planned. However, we took measures internally and there will be no customer impact, although the message will still appear.

Best wishes
Daniel

**Comment by MichaelThessel** [comment]
Thanks, much appreciated. Feel free to close the ticket if the issues are being addressed in the appropriate channels.

**Comment by h3raldo** [comment]
@danielneu - If Widgets.js has nothing to do with this module, where can I post a problem with Widgets.js? I was going to create a new issue here, but then I read your comment. (Getting  `D.indexOf is not a function` multiple times anywhere the script is included)

**Comment by danielneu** [comment]
Hi,

the Widgets.js file is used in any Amazon Pay integration. When included the right way and only once, it should make any issues for you.

My comment said the Widgets.js **size** is nothing which is related to this module.

Let's create a new issue for this, please add ways to reproduce (respectively an URL to check this) if possible.
